BungeeCord-Patches/0062-Add-root-admin-user-detection.patch

```diff
@@ -0,0 +1,77 @@
+From 763d9ab434a3fcf6dcf0ebd2218e55cbb22333e2 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Thu, 30 Sep 2021 16:59:18 +0200
+Subject: [PATCH] Add root/admin user detection
+
+This patch detects whether or not the server is currently executing as a privileged user and spits out a warning.
+The warning serves as a sort-of PSA for newer server admins who don't understand the risks of running as root.
+We've seen plenty of bad/malicious plugins hit markets, and there's been a few close-calls with exploits in the past.
+Hopefully this helps mitigate some potential damage to servers, even if it is just a warning.
+
+Co-authored-by: egg82 <eggys82@gmail.com>
+
+diff --git a/api/src/main/java/io/github/waterfallmc/waterfall/utils/ServerEnvironment.java b/api/src/main/java/io/github/waterfallmc/waterfall/utils/ServerEnvironment.java
+new file mode 100644
+index 00000000..1ec9fe05
+--- /dev/null
++++ b/api/src/main/java/io/github/waterfallmc/waterfall/utils/ServerEnvironment.java
+@@ -0,0 +1,34 @@
++package io.github.waterfallmc.waterfall.utils;
++
++import java.io.BufferedReader;
++import java.io.IOException;
++import java.io.InputStreamReader;
++
++public class ServerEnvironment {
++	private static final boolean RUNNING_AS_ROOT_OR_ADMIN;
++
++	static {
++		boolean isWindows = System.getProperty("os.name").startsWith("Windows");
```

Comment:
tabs 😢 

Does this implement the fixed check from the new PR?

Reply:
Java 8 doesn't seem to like the NTSystem/UnixSystem classes, so I used the "running a command" method from the start, so this isn't affected by the OpenJDK bug.